### ISSUE TITLE
CQL being deleted when going through formatter

### ISSUE THREAD

**Issue by jackcmeyer** [cqframework/clinical_quality_language#174]
When running a CQL file through the formatter, CQL code is being deleted when there are errors in the file.

For example, when running this CQL file: 
 
```
library test version '0.0.000'

using QDM

notvalueset "Female Administrative Sex": '2.16.840.1.113883.3.560.100.2'
valueset "TestValueset": '1.1.111'

context Patient

define "TestDefinition":
	["Encounter, Performed": "TestValueset "]
```

The output of the formatter is

```
library test version '0.0.000'

using QDM

valueset "TestValueset": '1.1.111'

context Patient

define "TestDefinition":
	["Encounter, Performed": "TestValueset "]
```

Another example is, when running this CQL file: 
```
library test version '0.0.000'

using QDM

valueset "TestValueset": '1.1.111'

context Patient

define "TestDefinition":
	['"Encounter, Performed": "TestValueset "]
```

The output of the formatter is: 

```
library test version '0.0.000'

using QDM

valueset "TestValueset": '1.1.111'

context Patient

define "TestDefinition":
```

Thoughts on this?

**Comment by brynrhodes** [comment]
Yes, when there is a syntax error in the input, the parser will not be able to construct a full parse tree, and since the formatter works by visiting the parse tree, any part of the parse tree that couldn't be constructed will be missed in the output.

We discussed this today and will add functionality to the formatter to detect the case that there are syntax errors resulting in partial parse trees and revert to just returning the input source, unformatted.

**Comment by brynrhodes** [comment]
This fix is included in the 1.2.11-SNAPSHOT release.
